src/raftstore/store/peer.rs

```diff
@@ -966,15 +965,25 @@ impl Peer {
             let region = snap_data.take_region();
             // For merge process, when applying snapshot or create new peer the stale source peer is destroyed asynchronously.
             // So here checks whether there is any overlap, if so, wait and do not handle raft ready.
-            if let Some(r) = region_ranges
-                .range((Excluded(enc_start_key(&region)), Unbounded::<Vec<u8>>))
+            // Note that: below logic can not find the overlap region with range (a, b] that a < snap_region.end_key < b,
+            // but it is okay cause this case would never happen.
+            // The most possible case is like this, but is abandoned in propose stage.
+            //     | region 1 |       region 2      |  store 1
+            //                   || split 2 into 3
+            //                   \/
+            //     | region 1 | region 2 | region 3 |
+            //                   || merge 2 into 1 (never happen, cause source region's log gap should not contain AdminRequest(split 2 into 3))
```

Comment:
What if merging `region 1` into `region 2`?

Reply:
`region 1` update merge target, and then `region 2` receives a new snapshot and decide to destroy `region 1`. And here logic can detect `region 1`'s range is overlapped with `region 2`'s snapshot, so wait here until the destroy of `region 1` finish.